src/common/rect.cpp

```diff
@@ -0,0 +1,40 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  A simple rectangle helper class.
+ */
+
+#include "src/common/rect.h"
+
+namespace Common {
+
+Rect::Rect() : x(0.0), y(0.0), w(0.0), h(0.0) {
+}
+
+Rect::operator bool() const {
+	return x != w && y != h;
```

Comment:
Also, are you sure that's correct?

So if you have this rectangle:

```
(5/5)      (10/5)
  .----------.
  |          |
  .__________.
(5/10)     (10/10)  
```

then

```
x = 5.0
y = 5.0
w = 5.0
h = 5.0
```

So this would return false.

This method would work if you'd use x1, y1, x2, y2, and not x, y, w, h. With width and height, you need to check for `w != 0.0 && h != 0.0`, no?